### PR TITLE
Update Kotlin and Gradle version

### PR DESCRIPTION
**Pull request by blasten**
Fixes conflict in https://github.com/flutter/flutter/pull/32550

Removed `FlutterEventLogger` since it doesn't rethrow Gradle failures anymore.

### CODE REVIEW

**Comment by blasten** [comment]
@ened would you be ok if I take over https://github.com/flutter/flutter/pull/32550?  If yes, please leave a comment that only contains `@googlebot I consent.`

**Comment by ened** [comment]
@googlebot I consent.

**Comment by ened** [comment]
@blasten, of course. Go crazy. :-)

File: dev/benchmarks/complex_layout/android/build.gradle [hunk 1]
```diff
@@ -5,7 +5,7 @@ buildscript {
     }
 
     dependencies {
-        classpath 'com.android.tools.build:gradle:3.2.1'
+        classpath 'com.android.tools.build:gradle:3.4.2'
```
Comment (dnfield) [inline dev/benchmarks/complex_layout/android/build.gradle]: For this and the gradle version - should we roll to latest?  If not, can you update the PR description to explain why we're not and whatever is blocking us from doing it?

File: packages/flutter_tools/gradle/flutter.gradle [hunk 2]
```diff
@@ -949,23 +949,3 @@ class FlutterPluginTask extends DefaultTask {
         }
     }
 }
-
-gradle.useLogger(new FlutterEventLogger())
```
Comment (dnfield) [inline packages/flutter_tools/gradle/flutter.gradle]: Could we get something in the PR description explaining why this was removed?  I'm assuming it uses deprecated APIs - what's the replacement (or why do we not need a replacement)?

**Review by dnfield — APPROVED** [review]
This LGTM - would appreciate clarifications in the PR description about why we're not moving to head on these versions (or just move them to head if there's nothing blocking it).

File: packages/flutter_tools/gradle/flutter.gradle (hunk 2 above)
Comment (blasten) [reply packages/flutter_tools/gradle/flutter.gradle]: Done. This was mentioned in #32550

Comment (blasten) [reply packages/flutter_tools/gradle/flutter.gradle]: Done

File: packages/flutter_tools/templates/app/android-kotlin.tmpl/build.gradle [hunk 3]
```diff
@@ -1,12 +1,12 @@
 buildscript {
-    ext.kotlin_version = '1.2.71'
+    ext.kotlin_version = '1.3.41'
     repositories {
```
Comment (ened) [inline packages/flutter_tools/templates/app/android-kotlin.tmpl/build.gradle]: This should be 1.3.50 now.

File: packages/flutter_tools/templates/plugin/android-kotlin.tmpl/build.gradle.tmpl [hunk 4]
```diff
@@ -2,14 +2,14 @@ group '{{androidIdentifier}}'
 version '1.0-SNAPSHOT'
 
 buildscript {
-    ext.kotlin_version = '1.2.71'
+    ext.kotlin_version = '1.3.41'
```
Comment (ened) [inline packages/flutter_tools/templates/plugin/android-kotlin.tmpl/build.gradle.tmpl]: As above

File: packages/flutter_tools/templates/app/android-kotlin.tmpl/build.gradle (hunk 3 above)
Comment (blasten) [reply packages/flutter_tools/templates/app/android-kotlin.tmpl/build.gradle]: Done

**Pull request merged**